Bottom: cefdd35d172882092a04a67cd6e438b0766ac974
Top:    08c26a514d917875e37088f2cdb9ed05d2ad9bfa
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-11-04 02:06:28 +0000

Test moving glEnableClientState calls

This might be better? Set up the pointers before enabling the state.


---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index b9b1e94..335e47d 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -146,14 +146,14 @@ hidgl_flush_triangles (triangle_buffer *buffer)
     buffer->triangle_array = NULL;
   }
 
-  glEnableClientState (GL_VERTEX_ARRAY);
   glVertexPointer (3, GL_FLOAT, 5 * sizeof (GLfloat), buffer->use_vbo ?
                      BUF_OFFSET (0) : buffer->triangle_array + 0);
 
-  glEnableClientState (GL_TEXTURE_COORD_ARRAY);
   glTexCoordPointer (2, GL_FLOAT, 5 * sizeof (GLfloat), buffer->use_vbo ?
                        BUF_OFFSET (3) : buffer->triangle_array + 3);
 
+  glEnableClientState (GL_VERTEX_ARRAY);
+  glEnableClientState (GL_TEXTURE_COORD_ARRAY);
   glDrawArrays (GL_TRIANGLE_STRIP, 0, buffer->vertex_count);
   glDisableClientState (GL_VERTEX_ARRAY);
   glDisableClientState (GL_TEXTURE_COORD_ARRAY);
